File: projects/two-way-chat/backend/Chat.Domain/Message.cs
namespace Chat.Domain;

/// <summary>
/// A single chat message. Immutable record — the server is the sole authority
/// for <see cref="Id"/> and <see cref="SentAt"/>; clients never supply these.
/// </summary>
public record Message(Guid Id, string RoomId, string SenderName, string Text, DateTimeOffset SentAt);

File: projects/two-way-chat/frontend/src/core/message.ts
/**
 * Mirrors the server's `Message` record (Chat.Domain/Message.cs). Field
 * order and names line up with the JSON SignalR sends over the wire.
 */
export interface Message {
  id: string;
  roomId: string;
  senderName: string;
  text: string;
  sentAt: string; // ISO 8601 string, server-stamped
}

export interface PresenceEvent {
  type: 'joined' | 'left';
  displayName: string;
}
